### PR TITLE
Fix infinite scrolling out of control

### LINKED ISSUE #2240: 🐛 [Bug] Infinite Scroll is loading more posts kinda infinitely after scrolling down
## :bug: Bugreport
The lazy loading does not load older articles, but seams to be stuck in some kind.

### Steps to reproduce the behavior
1. go to the root path
2. scroll down to trigger infinite scroll
3. see that it consistently scrolls without more scrolling

### Expected behavior
The lazy loading should load more articles in a manner respecting the currently set sorting order.

### Version & Environment
any

### PR DESCRIPTION
**Pull request by mattwr18**
### Issues

- fixes #2240

### CODE REVIEW

File: webapp/store/posts.js [hunk 1]
```diff
@@ -120,26 +114,4 @@ export const getters = {
...
-      variables: {
-        filter: getters.filter,
-        first,
-        orderBy: ['pinned_asc', getters.orderBy],
-        offset,
-      },
-    })
-    commit('SET_CURRENT_POSTS', Post)
-    return Post
-  },
```
[the remark this answers is not in the document]
Comment (mattwr18) [reply webapp/store/posts.js]: yeah, but how to do this when you are trying to reset the cache and refetch Posts from a `layout` to a `page`? I'm quite sure they don't have examples for this

File: webapp/package.json [hunk 2]
```diff
@@ -83,6 +83,7 @@
     "v-tooltip": "~2.0.2",
     "validator": "^12.0.0",
     "vue-count-to": "~1.0.13",
+    "vue-infinite-loading": "^2.4.4",
     "vue-infinite-scroll": "^2.0.2",
```
[the remark this answers is not in the document]
Comment (mattwr18) [reply webapp/package.json]: I don't want to remove `vue-infinite-loading`, I just added it cause it is being maintained, whereas `vue-infinite-scroll` is not.

[the remark this answers is not in the document]
Comment (mattwr18) [reply webapp/package.json]: I think I got the idea though... hehe

**Comment by roschaefer** [comment]
Ha, found a bug. Nothing that should stop us from deploying though:

Screenshot - 2019-11-15T152920 386

File: webapp/store/posts.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (roschaefer) [reply webapp/store/posts.js]: :thinking: local state is for state that is not stored in the backend. So `refetch` is certainly out of scope of the local state.

@mattwr18 why does the layout has to set posts for the page component :astonished: ?

[the remark this answers is not in the document]
Comment (mattwr18) [reply webapp/store/posts.js]: because they want to click on the logo which lives in the `default.vue` and it refetch the Posts...

**Pull request merged**